core.impl/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/impl/repair/scheduler/RepairSchedulerImpl.java

```diff
@@ -0,0 +1,395 @@
+        {
+            LOG.error("Interrupted while waiting for executor to shutdown", e);
+            Thread.currentThread().interrupt();
+        }
+
+        synchronized (myLock)
+        {
+            myScheduledJobs.entrySet().stream()
+                    .flatMap(nodeEntry -> nodeEntry.getValue().entrySet().stream()
+                            .flatMap(tableEntry -> tableEntry.getValue().stream()
+                                    .map(job -> new AbstractMap.SimpleEntry<>(nodeEntry.getKey(), job))
+                            )
+                    )
+                    .forEach(entry -> descheduleTableJob(entry.getKey(), entry.getValue()));
+
+            myScheduledJobs.clear();
+        }
+    }
+
+    @Override
+    public void putConfigurations(
+            final Node node,
+            final TableReference tableReference,
+            final Set<RepairConfiguration> repairConfiguration)
+    {
+        myExecutor.execute(() -> handleTableConfigurationChange(node, tableReference, repairConfiguration));
+    }
+
+    @Override
+    public void removeConfiguration(final Node node, final TableReference tableReference)
+    {
+        myExecutor.execute(() -> handleTableConfigurationRemoved(node, tableReference));
+    }
+
+    @Override
+    public List<ScheduledRepairJobView> getCurrentRepairJobs()
+    {
+        synchronized (myLock)
+        {
+            return myScheduledJobs.values().stream()
+                    .flatMap(tableJobs -> tableJobs.values().stream())
+                    .flatMap(Set::stream)
+                    .map(ScheduledRepairJob::getView)
+                    .collect(Collectors.toList());
+        }
+    }
+
+    private void handleTableConfigurationChange(
+            final Node node,
+            final TableReference tableReference,
+            final Set<RepairConfiguration> repairConfigurations)
+    {
+        synchronized (myLock)
+        {
+            try
+            {
+                if (configurationHasChanged(node, tableReference, repairConfigurations))
+                {
+                    LOG.info("Creating schedule for table {} in node {}", tableReference, node.getHostId());
+                    createTableSchedule(node, tableReference, repairConfigurations);
+                }
+                LOG.info("No configuration changes for table {} in node {}", tableReference, node.getHostId());
+            }
+            catch (Exception e)
+            {
+                LOG.error("Unexpected error during schedule change of {}:", tableReference, e);
+            }
+        }
+    }
+
+    private boolean configurationHasChanged(
+            final Node node,
+            final TableReference tableReference,
+            final Set<RepairConfiguration> repairConfigurations)
+    {
+        Set<ScheduledRepairJob> jobs = validateScheduleMap(node.getHostId(), tableReference);
+        if (jobs == null || jobs.isEmpty())
+        {
+            return true;
```

Comment:
I think this is actually never empty or null, because it is called in DefaulRepairConfigurationProvider when the schema changes, but I'm adding a validation to avoid returning true when it is null or empty

Reply:
But isn't that exactly what you're doing, "if null or empty > return true"?